handlers.go

```diff
@@ -104,15 +104,37 @@ func (m *MockOIDC) Token(rw http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	if !m.validateTokenParams(rw, req) {
-		return
-	}
-
-	code := req.Form.Get("code")
-	session, err := m.SessionStore.GetSessionByID(code)
-	if err != nil {
-		errorResponse(rw, invalidGrant, fmt.Sprintf("Invalid code: %s", code),
-			http.StatusUnauthorized)
+	var session *Session
+	grantType := req.Form.Get("grant_type")
+	switch grantType {
+	case "authorization_code":
+		if !m.validateTokenParams(rw, req) {
```

Comment:
I think we keep this common above, and just remove the inner `grant_type` validation it had.  We can handle when grant isn't one of our 2 types directly in the switch statement